src/main/java/com/onarandombox/MultiverseNetherPortals/listeners/MVNPEntityListener.java

```diff
@@ -49,41 +54,58 @@ public MVNPEntityListener(MultiverseNetherPortals plugin) {
         this.worldManager = this.plugin.getCore().getMVWorldManager();
         this.pt = new PermissionTools(this.plugin.getCore());
         this.playerErrors = new HashMap<String, Date>();
-        this.eventRecord = new HashMap<String, Location>();
         this.messaging = this.plugin.getCore().getMessaging();
         this.locationManipulation = this.plugin.getCore().getLocationManipulation();
-
+        this.eventRecord = new HashMap<>();
+        this.eventRecord.put(PortalType.ENDER, new HashSet<>());
+        this.eventRecord.put(PortalType.NETHER, new HashSet<>());
```

Comment:
This is fine. However, to me this smells like a good place for a purpose built class.

Reply:
I think that'd be fine. We could also make the event record static then, which I think would be good (let me know what you think).

Reply:
Nah. I like to avoid static for the most part with Bukkit stuff. If you use /reload it can lead to memory leaks.

Reply:
I see, good thing I asked.

Reply:
Just pushed it now 😄.